### PR TITLE
SCAL-192739 Added Number Formatting Utils

### PR DESCRIPTION
**Pull request by yashvardhan-n**

### CODE REVIEW

File: src/utils/globalize-Initializer/globalize-utils.ts [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    if (currentCurrencyFormat) {
+        return currentCurrencyFormat;
+    }
+
+    const countryCode = getCountryCode(currentLocale);
+    const regionData =
+        supplementalCurrencyDataJson?.supplemental?.currencyData?.region[
+            countryCode.toUpperCase()
+        ];
+    if (!regionData || regionData.length === 0) {
```
Comment (UtkarshRai12) [inline src/utils/globalize-Initializer/globalize-utils.ts:56]: pls change this to logger.warn already merged in sdk version 23.

File: src/utils/globalize-Initializer/globalize-utils.ts [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+        console.warn('No currency data found for country:', countryCode);
```
Comment (UtkarshRai12) [inline src/utils/globalize-Initializer/globalize-utils.ts:57]: pls change this to logger.warn.

File: src/utils/globalize-Initializer/globalize-utils.ts [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+ */
+export function formatNumberSafely<
+    FormatOptions extends Globalize.NumberFormatterOptions,
+>(format: FormatOptions, num: number): string {
+    try {
+        const formatter = globalizeNumberFormatter(format);
+        const formattedNumber = formatter(num);
+        return formattedNumber;
+    } catch (e) {
+        console.error('Error formatting pattern: ', format, num, e);
```
Comment (UtkarshRai12) [inline src/utils/globalize-Initializer/globalize-utils.ts:173]: pls add logger here also if u want this behind debug flag

File: src/utils/number-formatting/formatting-utils.ts [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+        value === strings.EMPTY_VALUE_PLACEHOLDER_LABEL
+    ) {
+        return value;
+    }
+
+    if (value === null || value === undefined) {
+        return strings.NULL_VALUE_PLACEHOLDER_LABEL;
+    }
+
+    if (value === Infinity || value === -Infinity || _.isNaN(value)) {
```
Comment (UtkarshRai12) [inline src/utils/number-formatting/formatting-utils.ts:254]: pls import `strings` from TS these are based on locale u can use it from `appConfig.tsLocaleBasedStringsFormats`. Also handle NaN using isNaN
```
export function isSpecialNaNValue(value) {
    const specialValue = getSpecialFormatData(value);
    if (!specialValue) {
        // eslint-disable-next-line no-restricted-globals
        return isNaN(value);
    }
    return specialValue === 'NaN';
}

File: src/utils/globalize-Initializer/globalize-utils.ts (hunk 2 above)
Comment (yashvardhan-n) [reply src/utils/globalize-Initializer/globalize-utils.ts:57]: ack

File: src/utils/globalize-Initializer/globalize-utils.ts (hunk 3 above)
Comment (yashvardhan-n) [reply src/utils/globalize-Initializer/globalize-utils.ts:173]: ack

File: src/utils/number-formatting/number-formatting.ts [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+        } catch (e) {
+            console.error(
+                'Corrupted format config passed, formatting using default config',
+                formatConfig,
+                e,
+            );
+        }
+    }
+
+    if (formatConfig.category === CategoryType.Custom) {
```
Comment (harshmeetTS) [inline src/utils/number-formatting/number-formatting.ts:214]: these if statements can be switch case

**Review by harshmeetTS — APPROVED** [review]

**Pull request merged**